### PR TITLE
ART-2134: add prepare-release job

### PR DESCRIPTION
**Pull request by vfreex**
This job will perform an assortment of weekly release tasks for creating a release from scratch. The job needs to know:

- the expected release name (e.g. 4.6.42)
- expected release date
- proposed nightlies for each arch

The tasks for preparing a release are:

- create advisories for a release
- update ocp-build-data for new advisory numbers and ensure the scheduled build is frozen.
- notify docs team to create live IDs
- create a release JIRA
- sweep bugs and builds into the advisories
- verify the swept builds match the nightlies
- notify QE teams (mainline and multiarch) about the proposed nightlies, advisories, and Jira

## Notes for reviewers
- The pipeline is implemented in Python (rather than calling existing Jenkins jobs), and the Jenkinsfile just takes job parameters, execute the python script, and save artifacts in the workspace.
- To make the JIRA automation easier, I've updated the templates https://issues.redhat.com/browse/OCPPLAN-4756 and https://issues.redhat.com/browse/OCPPLAN-1373 to follow [Jinja2](https://jinja.palletsprojects.com/) format. 
- Job parameters are minimum. Options rarely changing are stored in `config/artcd.toml`.
- This is my test build: https://saml.buildvm.openshift.eng.bos.redhat.com:8888/job/hack/job/yuxzhu/job/aos-cd-jobs/job/build%252Fprepare-release/19/console (it failed at the second last step because the complained images was removed from ocp-build-data but still exists in image streams, otherwise it would pass)
- Example command for running manually (add `--dry-run` if you don't really want to prepare a release):

```sh
# set credentials
$ kinit
$ oc login api.ci.openshift.org
$ oc registry login
$ docker login quay.io
$ bugzilla login --api-key
$ export JIRA_USERNAME=<your-jboss-jira-username> JIRA_PASSWORD=<your-jboss-jira-password>

# run the pipeline
$ pyartcd/prepare_release.py -vv --working-dir=./pyartcd_working --config=./config/artcd.toml 4.2.42 --package-owner yuxzhu@redhat.com --date 2020-Dec-09 --package-owner yuxzhu@redhat.com --nightly 4.2.0-0.nightly-2020-11-05-133907 --nightly 4.2.0-0.nightly-s390x-2020-11-05-133855
```
- I reordered some steps comparing with the existing jobs. Please take a look and see if the order is reasonable.

### CODE REVIEW

**Comment by vfreex** [comment]
docs https://github.com/openshift/aos-cd-jobs/pull/2438

File: pyartcd/pyartcd/pipelines/prepare_release.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+        ]
+        _LOGGER.debug("Running command: %s", cmd)
+        if self.dry_run:
+            _LOGGER.warn("Would have run: %s", cmd)
+        else:
+            subprocess.run(cmd, check=True, universal_newlines=True, cwd=self.working_dir)
+
+    def sweep_bugs(
+        self,
+        statuses: List[str] = ["ON_QA", "VERIFIED"],
```
Comment (sosiouxme) [inline pyartcd/pyartcd/pipelines/prepare_release.py:228]: updated today, can we do:
```suggestion
        statuses: List[str] = ["MODIFIED", "ON_QA", "VERIFIED"],
```

**Review by sosiouxme — COMMENTED** [review]
consolidate a bunch of stuff with some python eh?? well alright.
i suppose we should mark the `advisories` job deprecated and plan to remove it in a few months.

File: pyartcd/pyartcd/pipelines/prepare_release.py (hunk 1 above)
Comment (vfreex) [reply pyartcd/pyartcd/pipelines/prepare_release.py]: suggestion applied

**Review by thiagoalessio — APPROVED** [review]
/lgtm

**Pull request merged**